app/assets/javascripts/discourse/app/modifiers/d-autocomplete.js

```diff
@@ -330,6 +344,16 @@ export default class DAutocompleteModifier extends Modifier {
     this.previousTerm = term;
     this.searchTerm = term;
 
+    // Special handling for terms with spaces to allow for full name search (mentions only)
+    const hasSpaces = term.includes(" ") && term.trim().length > 0;
+    const isMentionAutocomplete = this.options.key === "@";
+
+    if (hasSpaces && isMentionAutocomplete) {
+      // Close menu but keep state for potential reopening and continue with search
+      this.pendingSpaceSearch = true;
+      await this.closeAutocomplete({ resetSearchState: false });
```

Comment:
I was playing around with it a little bit, how do you feel about only auto-closing immediately after a space? That seems like the most likely case where autocomplete should close, if it's still matching after a space has been typed, then we can assume that the user is typing something that should be matched. 

```suggestion
      if (term.slice(-1) === " ") {
        await this.closeAutocomplete({ resetSearchState: false });
      }
```